packages/flutter_test/lib/src/controller.dart

```diff
@@ -265,6 +335,304 @@ class SemanticsController {
 
     return false;
   }
+
+  /// Performs the given [SemanticsAction] on the [SemanticsNode] found by `finder`.
+  ///
+  /// If `args` are provided, they will be passed unmodified with the `action`.
+  /// The `checkForAction` argument allows for attempting to perform `action` on
+  /// `node` even if it doesn't report supporting that action. This is useful
+  /// for implicitly supported actions such as [SemanticsAction.showOnScreen].
+  void performAction(FinderBase<SemanticsNode> finder, SemanticsAction action, {Object? args, bool checkForAction = true}) {
+    final SemanticsNode node = finder.evaluate().single;
+    if (checkForAction && !node.getSemanticsData().hasAction(action)){
+      throw StateError(
+        'The given node does not support $action.\n'
+        'Node: $node'
+      );
+    }
+
+    node.owner!.performAction(node.id, action, args);
+  }
+
+  /// Performs a [SemanticsAction.tap] action on the [SemanticsNode] found
+  /// by `finder`
```

Comment:
nit (here and elsewhere):

```suggestion
  /// by `finder`.
```